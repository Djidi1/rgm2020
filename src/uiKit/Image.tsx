import React, { FC } from 'react';
import LazyLoad from 'react-lazyload';
import styled from 'styled-components';
import no_picture from '../img/No_Picture.jpg';

interface ImageProps {
  width?: string,
  height?: string,
}

type MainProps = {
  url: string,
};

const image: FC<MainProps> = ({ url, ...props }) => {
  const addDefaultSrc = (element) => {
    element.target.src = no_picture;
  };

  return (
    <LazyLoad height={320}>
      <img src={url} alt="img" onError={addDefaultSrc} {...props} />
    </LazyLoad>
  );
};

const Image = styled(image)<ImageProps>`
  width: ${({ width }) => width || '100%'};
  height: ${({ height }) => height || 'auto'};
`;

export default Image;
