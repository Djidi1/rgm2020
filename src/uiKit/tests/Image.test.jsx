import React from 'react';
import renderer from 'react-test-renderer';
import Image from '../Image';

describe('UI-Kit Image', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it('is expected to render successfully', () => {
    const wrapper = renderer
      .create(<Image />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('is expected to render successfully with params', () => {
    const wrapper = renderer
      .create(<Image width={100} height={100} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
