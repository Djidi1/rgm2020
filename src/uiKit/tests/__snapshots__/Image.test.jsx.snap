// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI-Kit Image is expected to render successfully 1`] = `
<div
  className="lazyload-wrapper"
>
  <div
    className="lazyload-placeholder"
    style={
      Object {
        "height": 320,
      }
    }
  />
</div>
`;

exports[`UI-Kit Image is expected to render successfully with params 1`] = `
<div
  className="lazyload-wrapper"
>
  <div
    className="lazyload-placeholder"
    style={
      Object {
        "height": 320,
      }
    }
  />
</div>
`;
